src/Components/Playlist_Modal/Playlist_Modal.jsx

```diff
@@ -1,18 +1,127 @@
-import React,{useState,useEffect} from 'react'
-import { useUserData } from '../../Context/UserData-Context';
+import React, { useState, useEffect } from "react";
+import { useUserData } from "../../Context/UserData-Context";
 import "./Playlist_Modal.css";
 import { useAuth } from "../../Context/Auth-Context";
+import toast, { Toaster } from "react-hot-toast";
 import axios from "axios";
 
 const Playlist_Modal = (props) => {
- 
+  const { modalVisible, modalVisibleHandler, currentVideo } = useUserData();
+  const [isChecked, setIsChecked] = useState("");
+  const [playlistName, setPlaylistName] = useState("");
+  const [playlistData, setPlaylistData] = useState([]);
+  const [renderHandler, setRenderHandler] = useState(true);
+  useEffect(() => {
+    (async () => {
+      const response = await axios.get("/api/user/playlists", {
+        headers: {
+          authorization: JSON.parse(localStorage.getItem("user")).encodedToken,
+        },
+      });
+      setPlaylistData(response.data.playlists);
+    })();
+  }, [renderHandler]);
+
+  const newPlaylistHandler = async () => {
+    const response = await axios.post(
+      "/api/user/playlists",
+      {
+        playlist: { title: playlistName, description: "bar bar bar" },
+      },
+      {
+        headers: {
+          authorization: JSON.parse(localStorage.getItem("user")).encodedToken,
+        },
+      }
+    );
+    setPlaylistName("");
+    setRenderHandler(!renderHandler);
+  };
+
+  const deletePlaylistHandler = async (playlistId) => {
+    const response = await axios.delete(`/api/user/playlists/${playlistId}`, {
+      headers: {
+        authorization: JSON.parse(localStorage.getItem("user")).encodedToken,
+      },
+    });
+    setRenderHandler(!renderHandler);
+  };
+
+  const addVideoHandler = async (playlistId) => {
+    const response = await axios.post(
+      `/api/user/playlists/${playlistId}`,
+      {
+        video: currentVideo,
+      },
+      {
+        headers: {
+          authorization: JSON.parse(localStorage.getItem("user")).encodedToken,
+        },
```

Comment:
You can extract localStorage as separate function as it is used in multiple places